lib/buyers/field.py

```diff
@@ -40,10 +48,35 @@ def to_python(self, value):
         # to save the hash, the same value we see is what is put in the DB.
         if not value.startswith(hasher.algorithm):
             self.salt = hasher.salt()
-            value = make_password(value, salt=self.salt)
+            value = self._hash(value, salt=self.salt)
         return HashedData(value)
 
 
+class ConsistentHashField(HashField):
+
+    """
+    Like the HashField, but is always a consistent value to allow lookups
+    on the field.
+    """
+
+    def get_prep_lookup(self, *args, **kw):
+        return super(CharField, self).get_prep_lookup(*args, **kw)
+
+    def _hash(self, value, salt=None):
+        """
+        Hash the value with hmac, using sha256. This ignores values of
+        salt, so that the hash can be consistent.
+        """
+        if not value:
+            return ''
+        if value.startswith('consistent:'):
```

Comment:
luckily an email address can't contain `:` otherwise this would be a security problem ;) Hopefully the next thing that uses `ConsistentSigField` gets lucky too.